client/src/App.tsx

```diff
@@ -39,8 +40,12 @@ const router = createBrowserRouter([
       },
       {
         path: "challenge",
-        element: <Challenge /> //TODO: REMOVE THIS ROUTE
-      }
+        element: <Challenge />, //TODO: REMOVE THIS ROUTE
+      },
+      {
+        path: "solutions/post/:challengeId",
```

Comment:
We should discuss how the API is going to be structured. I haven't really built any complex APIs in the past, but there should be guides and best practices.

Reply:
https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design

Reply:
Writing an OpenAPI specification might be a helpful step.

Reply:
Also for the links, I think we should structure around actions like /post_solution/:id instead of solutions/post/:id. I guess it's more of a stylistic decision.